src/adapters/dogs-unchained.ts

```diff
@@ -0,0 +1,81 @@
+// Copyright Abridged, Inc. 2022. All Rights Reserved.
+// Node module: @collabland/staking-contracts
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {BindingScope, extensionFor, injectable} from '@loopback/core';
+import {BigNumber} from 'ethers';
+import {STAKING_ADAPTERS_EXTENSION_POINT} from '../keys';
+import {BaseStakingContractAdapter, StakingAsset} from '../staking';
+import {DuStaking} from '../types/DuStaking';
+import {DuStaking__factory} from '../types/factories/DuStaking__factory';
+
+@injectable(
+  {
+    scope: BindingScope.SINGLETON,
+  },
+  extensionFor(STAKING_ADAPTERS_EXTENSION_POINT),
+)
+export class DogsUnchainedStakingContractAdapter extends BaseStakingContractAdapter {
+  contractAddress = '0xd742193c84062c1e0488545fb91a32d220ec6c76';
+  infoAddress = '0x5e2e43585df73a420b5866159f3de2a144c4d51e';
+
+  supportedAssets: StakingAsset[] = [
+    {
+      name: 'Dogs Unchained',
+      asset: 'ERC721:0x9c0ffc9088abeb2ea220d642218874639229fa7a',
+    },
+    {
+      name: 'Proof of Steak',
+      asset: 'ERC721:0xcab65c60d9dc47e1d450c7e9074f73f1ff75f181',
+    },
+    {
+      name: 'Puppies Unchained',
+      asset: 'ERC721:0x948bc723d7e33575427270ac4f26c73b8ba938aa',
+    },
+  ];
+
+  supplies: Record<string, number> = {
+    'Dogs Unchained': 9999,
+    'Proof of Steak': 1405,
+    'Puppies Unchained': 9999,
+  };
+
+  private async getOwnedAssets(
+    contract: DuStaking,
+    asset: StakingAsset,
+    owner: string,
+    maxId: number,
+  ) {
+    const token = asset.asset.replace('ERC721:', '');
```

Comment:
You can use the code below to pass `ERC721:<address>` as Chain Agnostic ID.

```ts
import {AssetName} from '@collabland/chain';
const token = new AssetName(asset.asset).reference;
```

Reply:
Updated! Heads up that tests passed on my end (checked linter as well), but looks like CI/CD is erroring on a missing key of some sort? Let me know if I can help resolve.

Reply:
Yes, `INFURA_PROJECT_ID` is configured as a github secret and it's invisible to pull requests.